updater/lib/dependabot/updater/operations/group_update_all_versions.rb

```diff
@@ -37,44 +35,48 @@ def initialize(service:, job:, dependency_snapshot:, error_handler:)
-              #
-              #        We could use all_updated_deps.first, but that could be misleading. It may
-              #        make more sense to handle the dependency group as a Dependancy-ish object
-              group_dependency = OpenStruct.new(name: "group-all")
-              raise if ErrorHandler::RUN_HALTING_ERRORS.keys.any? { |err| e.is_a?(err) }
-
-              error_handler.handle_dependabot_error(error: e, dependency: group_dependency)
+          dependency_snapshot.groups.each do |_group_hash, group|
+            Dependabot.logger.info("[Experimental] Starting grouped update job for #{job.source.repo}")
+            # We should log the rule being executed, let's just hard-code wildcard for now
+            # since the prototype makes best-effort to do everything in one pass.
+            # This should be replaced with the actual Dependabot::DependencyGroup instances that have names
+            # dependency_snapshot.groups will need to be updated to return a list of
+            # Dependabot::DependencyGroup instances
+            Dependabot.logger.info("Starting update group for '#{group.name}'")
+
+            dependency_change = compile_all_dependency_changes_for(group)
+
+            if dependency_change.updated_dependencies.any?
+              Dependabot.logger.info("Creating a pull request for '#{group.name}'")
+              begin
+                service.create_pull_request(dependency_change, dependency_snapshot.base_commit_sha)
+              rescue StandardError => e
+                # FIXME: This is a workround for not having a single Dependency to report against
+                #
+                #        We could use all_updated_deps.first, but that could be misleading. It may
+                #        make more sense to handle the dependency group as a Dependancy-ish object
+                group_dependency = OpenStruct.new(name: "group-all")
+                raise if ErrorHandler::RUN_HALTING_ERRORS.keys.any? { |err| e.is_a?(err) }
+
+                error_handler.handle_dependabot_error(error: e, dependency: group_dependency)
+              end
+            else
+              Dependabot.logger.info("Nothing to update for Dependency Group: '#{group.name}'")
             end
-          else
-            Dependabot.logger.info("Nothing to update for Dependency Group: '#{GROUP_NAME_PLACEHOLDER}'")
           end
-        end
-
-        private
 
-        attr_reader :job,
-                    :service,
-                    :dependency_snapshot,
-                    :error_handler,
-                    :dependency_group
+          # Let's not worry about this for now, but eventually we'd try to do single updates
+          # for anything that didn't appear in _at least one_ group.
+          dependency_snapshot.ungrouped_dependencies.each do |dependency|
+            # dependency_change = create_dependency_change_for(dependency)
+            # create_pull_request(dependency_change)
+          end
```

Comment:
```suggestion
          # FIXME: Let's not worry about this for now but eventually we'd try to do single updates
          # for anything that didn't appear in _at least one_ group.
          # dependency_snapshot.ungrouped_dependencies.each do |dependency|
          #   dependency_change = create_dependency_change_for(dependency)
          #   create_pull_request(dependency_change)
          # end
```

We might as well comment out the entire code block to avoid running the empty loop, I've been using `FIXME` as a search anchor for any pseduocode or notes I've been dropping in to make these things easy to find again.